plugins/imagebase/plugin.js

```diff
@@ -44,7 +66,84 @@
 		return CKEDITOR.plugins.link.parseLinkAttributes( widget.editor, widget.parts.link );
 	}
 
+	function createCaption( widget ) {
+		var element = widget.element,
+			caption = element.getDocument().createElement( 'figcaption' ),
+			definition = widget.editor.widgets.registered[ widget.name ];
+
+		element.append( caption );
+		widget.initEditable( 'caption', definition.editables.caption );
+
+		return caption;
+	}
+
+	function isEmptyOrHasPlaceholder( widget ) {
+
+		return !widget.editables.caption.getData() || widget.parts.caption.hasAttribute( 'data-cke-placeholder' );
+	}
+
+	function addPlaceholder( widget ) {
+		widget.parts.caption.setAttribute( 'data-cke-placeholder', true );
+		widget.editables.caption.setData( widget.editor.lang.imagebase.captionPlaceholder );
+	}
+
+	function removePlaceholder( widget ) {
+		widget.parts.caption.removeAttribute( 'data-cke-placeholder' );
+		widget.editables.caption.setData( '' );
+	}
+
 	var featuresDefinitions = {
+		caption: {
+			setUp: function( editor ) {
+				editor.on( 'selectionChange', function( evt ) {
+					var widgets = editor.widgets.instances,
+						i;
+
+					for ( i in widgets ) {
+						if ( widgets[ i ].features &&
+							CKEDITOR.tools.array.indexOf( widgets[ i ].features, 'caption' ) !== -1 ) {
+							widgets[ i ]._toggleCaption( evt.data.path.lastElement );
+						}
+					}
+				} );
+			},
+
+			init: function() {
+				if ( !this.parts.caption ) {
+					this.parts.caption = createCaption( this );
+				}
+
+				this._toggleCaption();
+			},
+
+			/**
+			 * Method used inside {@link CKEDITOR.editor#event-selectionChange} event to decide if caption for given widget
+			 * should be displayed and should contain placeholder text.
+			 *
+			 * @private
+			 * @member CKEDITOR.plugins.imagebase.featuresDefinitions.caption
+			 * @param {CKEDITOR.dom.element} sender Element, which triggered `selectionChange` event
+			 */
+			_toggleCaption: function( sender ) {
+				var isFocused = getFocusedWidget( this.editor ) === this,
+					caption = this.parts.caption,
+					editable = this.editables.caption;
+
+				if ( isFocused ) {
+					caption.removeAttribute( 'data-cke-hidden' );
```

Comment:
This attribute should be removed _after_ the placeholder was changed. This would mean less reflows which is always a good thing 👍